### PR TITLE
feat: Allow ECE Draft 06 headers.

### LINKED ISSUE #774: Update autopush to accept ECE Draft 06
This draft changes the content-type to "aes128gcm" & drops the "salt" from the crypto-key header.

see https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding-06

### PR DESCRIPTION
**Pull request by jrconlin**
NOTE: Not all clients support this draft yet. This change is being made
for developmental support.

closes #774

### CODE REVIEW

File: autopush/web/webpush.py [hunk 1]
```diff
@@ -204,6 +204,48 @@ def reject_encryption_key(self, data, original_data):
...
+        text.
+
+        """
+        dh = CryptoKey.parse_and_get_label(value, "dh")
+        if not dh or not VALID_BASE64_URL.match("dh"):
+            raise InvalidRequest("Invalid dh value in Encryption-Key header",
+                                 status_code=400,
+                                 errno=110)
+
+    @validates_schema(pass_original=True)
```
Comment (bbangert) [inline autopush/web/webpush.py]: Skip if field error?

Comment (jrconlin) [inline autopush/web/webpush.py]: that's copied from [WebPushCrypto04HeaderSchema](https://github.com/mozilla-services/autopush/pull/775/files/9784937e00d51921cead691a412dc63a77ee6590#diff-ccbbd52ff880cf4ddb71b18d126d189cR196) Should both be modified?

File: autopush/web/webpush.py [hunk 2]
```diff
@@ -204,6 +204,48 @@ def reject_encryption_key(self, data, original_data):
...
+class WebPushCrypto06HeaderSchema(Schema):
+    """Validate webpush Message Encryption
+
+    Uses draft-ietf-webpush-encryption-06 rules for validation
+
+    """
+    content_encoding = fields.String(
+        required=True,
+        load_from="content-encoding",
+        validate=OneOf(["aes128gcm"])
```
Comment (pjenvey) [inline autopush/web/webpush.py]: nitpick of something else copied from the other Schemas: this could be Equals(...) instead of OneOf([...])

**Comment by pjenvey** [comment]
+1 on delaying this as long as possible (until further standardized)

**Comment by jrconlin** [comment]
pjenvey gets his wish! The ECE spec is changing again. https://github.com/martinthomson/encrypted-content-encoding/pull/31 closing this PR.

**Pull request closed without merging**